File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
from collections import Counter
def solution(top):
    ans = 0
    c = Counter(top)
    s = set()
    for t in top:
        s.add(t)
        c[t] -= 1
        if c[t] == 0:
            del c[t]
        if len(s) == len(c): ans += 1
    return ans

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.kt
class Solution {
    fun solution(topping: IntArray): Int {
        var ans = 0
        val cnt = mutableMapOf<Int,Int>()
        topping.forEach {
            cnt[it] = cnt.getOrDefault(it,0) + 1
        }
        val s = mutableSetOf<Int>()
        for (t in topping) {
            s.add(t)
            cnt[t] = cnt[t]!! - 1
            if (cnt[t] == 0) {
                cnt.remove(t)
            }
            if (cnt.size == s.size) ans++
        }
        return ans
    }
}